README.md

```diff
@@ -1 +1,238 @@
+
+```js
+fs.writeFileSync(filename, content, options)
+```
+
+If we add console logs, the output will also has "Start writing file" and " Finish writing file"
+
+```js
+var filename = './fake-path/input.txt'
+var content = fs.readFileSync(filename, options)
+var fileToWriteTo = './fake-path/output.txt'
+console.log('Start writing file') 
+fs.writeFileSync(filToWriteTo, content, options)
+console.log('Finish writing file')
+```
+
+`fs` has other synchronous methods but the idea is usually the same—you get the data as the result of the expression and the next line is executed **after** the synchronous method is done.
+
+In most cases, we are dealing with concurrency and so we must use asynchronous code to make our apps performant. Nevertheless, you should know that it's possible so write synchronous code and when it's appropriate.
+
+
+## Async Way of Thinking
+
+To warm up with the async code, consider an example in which we have a few `console.log` statements and a timeout:
+
+```js
+var a = 1
+console.log('The value of a is ', a)
+setTimeout(function(){
+  a = 2
+  console.log('While the value of a is', a)
+}, 500)
+console.log('Now the value of a is', a)
+```
+
+For someone not versed in async way of thinking, it might appear that the resulting output will be
+
+```
+The value of a is  1
+While the value of a is 2
+Now the value of a is 2
+```
+
+Because in most languages the execution goes in the order of the statements, a person might think that because `a =2` precedes `Now the value...`, the second and third output will have the values of 2. This is incorrect, because `setTimeout()` is scheduled asynchronously. The right output will be:
+
+```js
+The value of a is  1
+Now the value of a is 1
+// delay of roughly 500 milliseconds
+While the value of a is 2
+```
+
+This misunderstanding of async code can lead to many bugs. It's important to pay attention to how async works to avoid creating issues for yourself down the line.
+
+
+## fs.readFile
+
+Let's take our synchronous code to read a file, and translate it to asynchronous:
+
+```js
+var fs = require('fs')
+console.log('Start reading file')
+var content = fs.readFileSync('readme.md', 'utf-8')
+console.log('Finish reading file')
+console.log(content)
+```
+
+The code will produce "Start reading file", then wait until the file reading is over and finally print "Finish reading file".
+
+The async method `fs.readFile` is a better because our program can do something else while it waits for the content of the file. There are a few things which are different in `fs.readFile` comparing to `fs.readFileSync`:
+
+1. We need to use callback
+2. The callback accepts an error object as a parameter
+3. The callback accepts the content object (the data from the file) as a parameter
+4. We don't get the content right away because `readFile` is **not an expression** like `fs.readFileSync`. (Expressions are functions which return something.) In other words, we cannot write `var content = fs.readFile(...)`.
```

Comment:
I don't think we need to define "expression" here -- I think point 3 covers what the students need to know. Point 4 can just indicate that `fs.readFile` doesn't return anything (it returns `undefined`).